src/layouts/EditHomepage.jsx

```diff
@@ -97,189 +97,222 @@ const enumSection = (type, isErrorConstructor) => {
-        } = await getSiteColors(siteName)
-
-        primaryColor = sitePrimaryColor
-        secondaryColor = siteSecondaryColor
-
-        if (this._isMounted) setSiteColors((prevState) => ({
-          ...prevState,
-          [siteName]: {
-            primaryColor,
-            secondaryColor,
-          }
-        }))
-      } else {
-        primaryColor = siteColors[siteName].primaryColor
-        secondaryColor = siteColors[siteName].secondaryColor
+        createPageStyleSheet(siteName, primaryColor, secondaryColor)
+      
+      } catch (err) {
+        console.log(err);
       }
 
-      createPageStyleSheet(siteName, primaryColor, secondaryColor)
-    
-    } catch (err) {
-      console.log(err);
-    }
+      try {
+        const resp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/homepage`, {
+          withCredentials: true,
+        });
+        const { content, sha } = resp.data;
+        const base64DecodedContent = Base64.decode(content);
+        const { frontMatter } = frontMatterParser(base64DecodedContent);
+        // Compute hasResources and set displaySections
+        let hasResources = false;
+        const displaySections = [];
+        let displayHighlights = [];
+        let displayDropdownElems = [];
+        const sectionsErrors = [];
+        let dropdownElemsErrors = [];
+        let highlightsErrors = [];
+        frontMatter.sections.forEach((section) => {
+          // If this is the hero section, hide all highlights/dropdownelems by default
+          if (section.hero) {
+            const { dropdown, key_highlights: keyHighlights } = section.hero;
+            if (dropdown) {
+              // Go through section.hero.dropdown.options
+              displayDropdownElems = _.fill(Array(dropdown.options.length), false);
+              // Fill in dropdown elem errors array
+              dropdownElemsErrors = _.map(dropdown.options, () => DropdownElemConstructor(true));
+              // Fill in sectionErrors for hero with dropdown
+              sectionsErrors.push({
+                hero: {
+                  title: '', subtitle: '', background: '', button: '', url: '', dropdown: '',
+                },
+              });
+            }
+            if (keyHighlights) {
+              displayHighlights = _.fill(Array(keyHighlights.length), false);
+              // Fill in highlights errors array
+              highlightsErrors = _.map(keyHighlights, () => KeyHighlightConstructor(true));
+              // Fill in sectionErrors for hero with key highlights
+              sectionsErrors.push({
+                hero: {
+                  title: '', subtitle: '', background: '', button: '', url: '',
+                },
+              });
+            }
+            if (!dropdown && !keyHighlights) {
+              sectionsErrors.push({
+                hero: {
+                  title: '', subtitle: '', background: '', button: '', url: '',
```

Comment:
nit: there is some repetition on the `sectionsErrors.hero` attribute. Maybe we can do something like:

```
const hero = { title: '', subtitle: '', background: '', button: '', url: '' }

if (dropdown) hero.dropdown = ''
if (keyHighlights) <blah>

sectionsErrors.push({ hero })

Reply:
Good point, resolved in https://github.com/isomerpages/isomercms-frontend/pull/334/commits/a1c4f55b818f4d2f66c353b5c67fa5d9016f3068